server/src/main/java/org/apache/iotdb/db/query/udf/core/transformer/CompareEqualToTransformer.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.query.udf.core.transformer;
+
+import org.apache.iotdb.db.query.udf.core.reader.LayerPointReader;
+
+public class CompareEqualToTransformer extends CompareOperatorTransformer {
+
+  public CompareEqualToTransformer(
+      LayerPointReader leftPointReader, LayerPointReader rightPointReader) {
+    super(leftPointReader, rightPointReader);
+  }
+
+  @Override
+  protected double evaluate(double leftOperand, double rightOperand) {
+    return leftOperand == rightOperand ? 1.0 : 0.0;
```

Comment:
It's not suggested that use `==` to compare double directly, as there may be precision loss. 
Please reference the review comment of `CompareOperatorTransformer` and change like this

```suggestion
protected boolean evaluate(boolean leftOperand, boolean rightOperand) {
      return Double.compare(leftOperand, rightOperand) == 0;
}